ramanujan/poly_domains/CartesianProductPolyDomain.py

```diff
@@ -89,10 +90,53 @@ def iter_polys(self, primary_looped_domain):
 
 	def get_a_coef_iterator(self):
 		return product(*self.an_domain_range)
-	
+
 	def get_b_coef_iterator(self):
 		return product(*self.bn_domain_range)
-	
+
 	def get_individual_polys_generators(self):
 		# for backwards compatibility.
 		return self.get_a_coef_iterator(), self.get_b_coef_iterator()
+
+	@staticmethod
+	def _get_metadata_on_var_ranges(ranges, series):
+		ranges_metadata = []
+		for i, v in enumerate(ranges):
+			ranges_metadata.append({
+				'range': v,
+				'size': v[1]-v[0]+1,
+				'series': series,
+				'index': i
+				})
+		return ranges_metadata
+
+	def split_domains_to_processes(self, number_of_instances):
```

Comment:
please add a unit test for this function, making sure that the split is done correctly - 
all elements exist in some partition, and  no more than the total number of elements expected